docs/FOSS_puppet_server.md

```diff
@@ -1,62 +1,104 @@
 
 - [example42 PSICK Puppet Open Source Server automation](#example42-psick-puppet-open-source-server-automation)
     - [1. Vagrant installation](#1-vagrant-installation)
+        - [Prerequisites](#prerequisites)
+        - [Procedure](#procedure)
     - [2. Base OS installation](#2-base-os-installation)
+        - [Prerequisites](#prerequisites-1)
+        - [Procedure](#procedure-1)
 
 ## example42 PSICK Puppet Open Source Server automation
 
 This `control-repo` allows to spin up a `Puppet Open Source Server` in a fully automated way.
 
 One can use [vagrant](https://www.vagrantup.com/) or even use `PSICK` on a fresh base OS installation.
 
-First we need `puppet-agent` and `r10k` installed.
 
 ### 1. Vagrant installation
 
-For ```vagrant``` it is required to have a local `ruby` installation. Best option is to either use [rvm](https://rvm.io/) or [rbenv](https://github.com/rbenv/rbenv). Then run:
+#### Prerequisites
+
+  - `Vagrant`
+  - local `ruby` installation. Best option is to use [rvm](https://rvm.io/) or [rbenv](https://github.com/rbenv/rbenv)
+  - root rights
+  - `puppet-agent` and `r10k` packages installed
+
+
+#### Procedure
+
+Install either [rvm](https://rvm.io/) or [rbenv](https://github.com/rbenv/rbenv). Then run:
+
 ```
     bundle install --path vendor/bundle
 ```
 
 Use ```bundle``` to install the modules via ```r10k``` from ```Puppetfile```:
 
+```
     bundle exec r10k puppetfile install -v
+```
 
-Now change to ```vagrant/environments/pos``` directory and run ```vagrant up puppet.foss.psick.io```.
+Now change to ```vagrant/environments/foss``` directory and start up `Vagrant` machine.
+
+```
+    cd vagrant/environments/foss
+    vagrant up puppet.foss.psick.io
+```
 
 ### 2. Base OS installation
 
-To install the `FOSS Puppet Server` with `PuppetDB` on a vanilla OS (tested ```RedHat 7``` derivatives and ```Ubuntu 16.04```) the easiest way is cloning `control-repo` into ```/etc/puppetlabs/code/environments/production``` folder and apply the **puppet_foss_master** role:
+#### Prerequisites
+
+  - at least 4 GB of RAM to run `Puppet Server` and `PuppetDB` (with the `PostgreSQL` backend) on the same node
+  - compatible OS (tested on ```RedHat 7``` derivatives and ```Ubuntu 16.04```)
+  - root rights
+  - `puppet-agent` and `r10k` packages installed
+
 
+#### Procedure
+
+The easiest way to install the `FOSS Puppet Server` with `PuppetDB` on a vanilla OS is to clone `control-repo` into ```/etc/puppetlabs/code/environments/production``` folder and apply the **```puppet_foss_master```** `role`:
+
+```
+    sudo -s
+ 
     mkdir -p /etc/puppetlabs/code/environments
     cd /etc/puppetlabs/code/environments
     git clone https://github.com/example42/psick.git production
     cd production
 
-Next it is required to have `puppet-agent` package already installed. We can install `Puppet 5 agent` running from `PSICK` base directory:
-
-    bin/puppet_install.sh
+    PSICK_DIR=$(git rev-parse --show-toplevel)
```

Comment:
For me, it was very confusing to understand where commands should be run from so `shell variable` should solve this issue. 

I also like the `control_repo_dir` naming like you did in [git.md](https://github.com/example42/psick/blob/5d4609199aeecde9874c6138c5018b44ea452685/docs/git.md#L212)

which shell variable do you prefer more?  `PSICK_DIR` or `CONTROL_REPO_DIR`? the latter is bit long but makes more sense.

Reply:
Usually all the commands are supposed to be run from the control repo main dir, unless stated otherwise. And, yes, CONTROL_REPO_DIR is more appropriate, maybe CONTROLREPO_DIR looks better, but really don't have any real preference here.